src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -124,6 +124,11 @@
   {:name schema/Str
    :files [EnvironmentClassesFileEntry]})
 
+(def EnvironmentModulesInfo
+  "Schema for the return payload an environment_classes request."
+  {:name schema/Str
+   :modules schema/Any}) ; TODO: fixme
```

Comment:
I think the problem there is that while `RubyHash` implements `java.util.Map`, schema only seems to work well with Clojure map classes, like `clojure.lang.PersistentArrayMap`.

For the environment_classes endpoint, I wrote a function in the `master_core` namespace called `sort-nested-environment-class-info-maps`.  I used that to transform the Java class types coming from JRuby into Clojure-friendly types that work with Clojure's `sorted-map`, which I used to preserve element order for serialization purposes.  This has the benefit of schema maps being usable to validate the output as well.  Serialization order of map elements is more crucial for the environment_classes endpoint since we calculate etags for use with its caching layer and need those to be consistent for the whole payload.  The environment_modules endpoint doesn't need to deal with etags currently, so the extra ordering benefit for serialization isn't critical.  Assuming a little extra memory usage and a little poorer performance isn't a big deal - which I believe to be true at this point - I think using this function could help for schema validation here.

For example, I got this to work:

```clj
(def EnvironmentModuleInfo
  {:forge_name (schema/maybe schema/Str)
   :name schema/Str
   :version schema/Str})

(def EnvironmentModulesInfo
   "Schema for the return payload an environment_classes request."
   {:name schema/Str
    :modules [EnvironmentModuleInfo]})

(schema/defn ^:always-validate
  module-info-from-jruby->module-info-for-json  :- EnvironmentModulesInfo
   [info-from-jruby :- List
    environment :- schema/Str]
   (->> info-from-jruby
        sort-nested-environment-class-info-maps
        vec
        (sorted-map :name environment :modules)))
```

Assuming you want to go this way, I think it would be good to rename `sort-nested-environment-class-info-maps` to be something more generic like `sort-nested-java-maps` or something.